pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -102,6 +103,15 @@ var (
 
 	// accessLogBuilder is used to set accessLog to filters
 	accessLogBuilder = newAccessLogBuilder()
+
+	// accessLogFormatters configures additional formatters needed for some of the format strings like "REQ_WITHOUT_QUERY"
+	accessLogFormatters = []*core.TypedExtensionConfig{
```

Comment:
shouldn't we only add it if they use `REQ_WITHOUT_QUERY`? Otherwise we add this filter to everyone when 99.9% of users do not use it?

Reply:
I initially did that but later removed it because the check.is not needed as this won't add any overhead and envoy will use it if the format has that field

Reply:
Well it does have overhead as the XDS will become even more bloated. The access logs are one of the biggest parts of XDS already

Reply:
I am not sure if this formatter array would cause significant bloating. But good to add it only when needed. Added PTAL.